packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -170,6 +265,75 @@ export class GarbageCollector implements IGarbageCollector {
 
         // Whether we are running in test mode. In this mode, unreferenced nodes are immediately deleted.
         this.testMode = getLocalStorageFeatureGate(gcTestModeKey) ?? gcOptions.runGCInTestMode === true;
+
+        // Get the GC state from the GC blob in the base snapshot. Use LazyPromise because we only want to do
+        // this once since it involves fetching blobs from storage which is expensive.
+        const baseSummaryStateP = new LazyPromise<IGCState>(async () => {
+            const gcState: IGCState = { gcNodes: {} };
+            if (baseSnapshot === undefined) {
+                return gcState;
+            }
+
+            // Get GC blobs from each data store's summary tree. Get them and consolidate into IGCState format.
+            const dataStoreSnaphotTree = getSummaryForDatastores(baseSnapshot, metadata);
+            assert(dataStoreSnaphotTree !== undefined, "Expected data store snapshot tree in base snapshot");
+            for (const [dsId, dsSnapshotTree] of Object.entries(dataStoreSnaphotTree.trees)) {
+                const blobId = dsSnapshotTree.blobs[gcBlobName];
+                if (blobId === undefined) {
+                    continue;
+                }
+
+                const gcSummaryDetails = await readAndParseBlob<IGarbageCollectionSummaryDetails>(blobId);
+                // If there are no nodes for this data store, skip it.
+                if (gcSummaryDetails.gcData?.gcNodes === undefined || gcSummaryDetails.gcData.gcNodes === {}) {
+                    continue;
+                }
+                const dsRootId = `/${dsId}`;
```

Comment:
It looks like code above this line does not depend on dsId / dsSnapshotTree, so it should be moved outside of for loop?